app/next-client-app/types/dataset.ts

```diff
@@ -26,4 +26,15 @@ interface DataSetSRList {
   viewers: number[];
   admins: number[];
   editors: number[];
+  projects: number[];
+}
+interface Projects {
+  id: number;
+  name: string;
+  members: number[];
+}
+
+interface Users {
```

Comment:
```suggestion
interface User {
```
Similarly this represents just one User, not many